From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Thu, 25 Sep 2025 11:42:32 +0200
Subject: test(SKIPCPIO): introduce cpio_list_first function

In preparation to support 3cpio as alternative archive tool, move the
cpio listing call into `cpio_list_first` and expect it to take a file as
input.

Forwarded: https://github.com/dracut-ng/dracut-ng/pull/1705
---
 test/TEST-81-SKIPCPIO/test.sh | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/test/TEST-81-SKIPCPIO/test.sh b/test/TEST-81-SKIPCPIO/test.sh
index 630aece..99e6774 100755
--- a/test/TEST-81-SKIPCPIO/test.sh
+++ b/test/TEST-81-SKIPCPIO/test.sh
@@ -10,6 +10,11 @@ test_check() {
     (command -v cpio && command -v find && command -v diff) &> /dev/null
 }
 
+cpio_list_first() {
+    local file="$1"
+    cpio --extract --quiet --list < "$file"
+}
+
 skipcpio_simple() {
     mkdir -p "$CPIO_TESTDIR/skipcpio_simple/first_archive"
     pushd "$CPIO_TESTDIR/skipcpio_simple/first_archive"
@@ -32,7 +37,7 @@ skipcpio_simple() {
         | cpio -o --null -H newc >> "$CPIO_TESTDIR/skipcpio_simple.cpio"
     popd
 
-    cpio -i --list < "$CPIO_TESTDIR/skipcpio_simple.cpio" \
+    cpio_list_first "$CPIO_TESTDIR/skipcpio_simple.cpio" \
         > "$CPIO_TESTDIR/skipcpio_simple.list"
     cat << EOF | diff - "$CPIO_TESTDIR/skipcpio_simple.list"
 .
@@ -47,7 +52,8 @@ EOF
         skipcpio_path="${PKGLIBDIR}"
     fi
     "$skipcpio_path"/skipcpio "$CPIO_TESTDIR/skipcpio_simple.cpio" \
-        | cpio -i --list > "$CPIO_TESTDIR/skipcpio_simple.list"
+        > "$CPIO_TESTDIR/skipped.cpio"
+    cpio_list_first "$CPIO_TESTDIR/skipped.cpio" > "$CPIO_TESTDIR/skipcpio_simple.list"
     cat << EOF | diff - "$CPIO_TESTDIR/skipcpio_simple.list"
 .
 10
